tests/execution_engine/test_sqlalchemy_execution_engine.py

```diff
@@ -1003,3 +1059,159 @@ def validate_tmp_tables(execution_engine):
+class TestExecuteQuery:
+    """What does this test and why?
+
+    The `SQLAlchemyExecutionEngine.execute_query()` method exists to encapsulate
+    the creation and management of connections. Here we check that it works
+    as intended.
+    """
+
+    def test_execute_query(self, sa, pd_dataframe: pd.DataFrame):
+        execution_engine: SqlAlchemyExecutionEngine = build_sa_execution_engine(
+            pd_dataframe, sa
+        )
+
+        select_all = "SELECT * FROM test;"
+        result = execution_engine.execute_query(sa.text(select_all)).fetchall()
+
+        expected = [(1, 4), (2, 4)]
+        assert [r for r in result] == expected
+
+    def test_execute_query_in_transaction(self, sa, pd_dataframe: pd.DataFrame):
+        execution_engine: SqlAlchemyExecutionEngine = build_sa_execution_engine(
+            pd_dataframe, sa
+        )
+
+        select_all = "SELECT * FROM test;"
+        result = execution_engine.execute_query_in_transaction(
+            sa.text(select_all)
+        ).fetchall()
+
+        expected = [(1, 4), (2, 4)]
+        assert [r for r in result] == expected
+
+
+@pytest.mark.unit
+class TestConnectionPersistence:
+    """What does this test and why?
+
+    sqlite/mssql temp tables only persist within a connection, so we need to keep the connection alive.
+    These tests ensure that we use the existing connection if one is available.
+    """
+
+    def test_same_connection_used_from_static_pool_sqlite(
+        self, sa, pd_dataframe: pd.DataFrame
+    ):
+        """What does this test and why?
+
+        We want to make sure that the same connection is used for subsequent queries for databases that need it e.g.
+        sqlite.
+        Here we test that by creating a temp table and then querying it multiple times (each time pulling a connection
+        from the pool). The same connection should be pulled from the pool, if the connection wasn't the same the
+        temporary table wouldn't be accessible.
+        """
+        execution_engine = SqlAlchemyExecutionEngine(connection_string="sqlite://")
+        with execution_engine.get_connection() as con:
+            add_dataframe_to_db(df=pd_dataframe, name="test", con=con, index=False)
+
+        assert (
+            execution_engine.dialect_name == GXSqlDialect.SQLITE
+        ), "Error here means test setup failed."
+
+        create_temp_table = "CREATE TEMPORARY TABLE temp_table AS SELECT * FROM test;"
+        execution_engine.execute_query_in_transaction(sa.text(create_temp_table))
+
+        select_temp_table = "SELECT * FROM temp_table;"
+
+        res = execution_engine.execute_query(sa.text(select_temp_table)).fetchall()
+        res2 = execution_engine.execute_query(sa.text(select_temp_table)).fetchall()
+
+        # This assert is here just to make sure when we assert res == res2 we are not comparing None == None
+        expected = [(1, 4), (2, 4)]
+        assert [r for r in res] == expected
+
+        assert res == res2
```

Comment:
@anthonyburdi I might just be missing this, but which part tests that the `con` is the same as the one used from the static pool?

Reply:
Updated and added in https://github.com/great-expectations/great_expectations/pull/7607/commits/a1a8b07371490af399011b6820265c620c78d006